applications/StructuralMechanicsApplication/tests/test_adjoint_analysis_nonlinear_truss_structure.py

```diff
@@ -0,0 +1,115 @@
+        for i in range(1,6):
+            node_sensitivity.append( model_part_adjoint.GetNode(i).GetSolutionStepValue(KratosMultiphysics.SHAPE_SENSITIVITY) )
+
+        # ## Testing Adjoint sensitivity values with Finite difference results
+        self.assertAlmostEqual(node_sensitivity[0][0],-0.056284024907427004, 0)
+        self.assertAlmostEqual(node_sensitivity[0][1], -0.056284024907427004, 3)
+        self.assertAlmostEqual(node_sensitivity[1][0], 1.932646787583536, 2)
+        self.assertAlmostEqual(node_sensitivity[1][1], -1.9382867385342448, 2)
+        self.assertAlmostEqual(node_sensitivity[2][0], -2.5592816343511515, 2)
+        self.assertAlmostEqual(node_sensitivity[2][1], 1.9672906711676317, 1)
+        
+        self._removeH5Files("primal_output_truss")
+
+    def testThreeDimensionalStructure(self):
+        with open("response_function_tests/adjoint_strain_energy_response_parameters_3D_truss.json",'r') as parameter_file:
+            parameters = KratosMultiphysics.Parameters( parameter_file.read())
+
+        model = KratosMultiphysics.Model()
+        response_function = structural_response_function_factory.CreateResponseFunction("dummy", parameters["kratos_response_settings"], model)
+        model_part_adjoint = response_function.adjoint_model_part
+        response_function.RunCalculation(calculate_gradient=True)
+        
+        node_sensitivity = []
+        for i in range(1,6):
+            node_sensitivity.append( model_part_adjoint.GetNode(i).GetSolutionStepValue(KratosMultiphysics.SHAPE_SENSITIVITY) )
+
+        ## Testing Adjoint sensitivity values with Finite difference results
+        self.assertAlmostEqual(node_sensitivity[0][0], 0.003973418927216699, 2)
+        self.assertAlmostEqual(node_sensitivity[0][1], 0.00012844054708693875, 2)
+        self.assertAlmostEqual(node_sensitivity[0][2], 12.575784792545617, 2)
+        self.assertAlmostEqual(node_sensitivity[1][0], -6.742208012155969, 1)
+        self.assertAlmostEqual(node_sensitivity[1][1], -3.371206929614345, 1)
+        self.assertAlmostEqual(node_sensitivity[1][2], -3.143893783885687, 2)
+        self.assertAlmostEqual(node_sensitivity[2][0], 6.742758621669508, 1)
+        self.assertAlmostEqual(node_sensitivity[2][1], -3.3712069225089176, 1)
+        self.assertAlmostEqual(node_sensitivity[2][2], -3.143893783885687, 2)
+
+        self._removeH5Files("primal_output_truss")
+
+    def testMisesTrussStructure(self):
+        with open("response_function_tests/adjoint_strain_energy_response_parameters_truss.json",'r') as parameter_file:
+            parameters = KratosMultiphysics.Parameters( parameter_file.read())
+
+        model = KratosMultiphysics.Model()
+        response_function = structural_response_function_factory.CreateResponseFunction("dummy", parameters["kratos_response_settings"], model)
+        model_part_adjoint = response_function.adjoint_model_part
+        response_function.RunCalculation(calculate_gradient=True)
```

Comment:
Hm this does not follow the notation of the AnalysisStage

Reply:
Sorry, but I don't understand the question

Reply:
The test takes about three seconds

Reply:
take a look at the AnalysisStage and the python-solvers in StructuralMechanics

=> maybe RunSolution would be a better name
@armingeiser ?

Reply:
RunSolution() would be ok for me, however this is not related to this PR because its already done like this in the existing responses.